lua/mason-core/spawn.lua

```diff
@@ -65,12 +65,13 @@ setmetatable(spawn, {
 
             -- Find the executable path via vim.fn.exepath on Windows because libuv fails to resolve certain executables
             -- in PATH.
-            if platform.is.win and (spawn_args.env and has_path(spawn_args.env)) == nil then
+            if platform.is.win and (#canonical_cmd <= 4 or canonical_cmd:lower():sub(-4) ~= ".exe") then
```

Comment:
This condition seems very arbitrary to me, why only for command names `<= 4` characters long? Also, mason never calls `spawn` with `.exe` explicitly so the second condition is unnecessary.